main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -285,11 +285,16 @@ object CompletionProvider {
   private def getSyntacticContext(uri: String, pos: Position, errors: List[CompilationMessage]): SyntacticContext =
     errors.filter({
       case err => pos.line <= err.loc.beginLine
-    }).collectFirst({
-      case ParseError(_, ctx, _) => ctx
-      case WeederError.MalformedIdentifier(_, _) => SyntacticContext.Import
-      case ResolutionError.UndefinedType(_, _, _) => SyntacticContext.Type.OtherType
-      case ResolutionError.UndefinedName(_, _, _, isUse, _) => if (isUse) SyntacticContext.Use else SyntacticContext.Expr.OtherExpr
-    }).getOrElse(SyntacticContext.Unknown)
+    }).map({
```

Comment:
Can't this whole thing just be a minByOption?

Reply:
Assumes non-emptiness.